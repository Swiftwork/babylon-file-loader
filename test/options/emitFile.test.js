/* eslint-disable
  prefer-destructuring,
*/
import webpack from '../helpers/compiler';

describe('Options', () => {
  describe('emitFile', () => {
    describe('{Boolean}', () => {
      test('True (Default)', async () => {
        const config = {
          loader: {
            test: /\.babylon$/,
            options: {
              emitFile: true,
            },
          },
        };

        const stats = await webpack('fixture.js', config);
        const { assets } = stats.toJson().modules[1];

        expect(assets[0]).toMatchSnapshot();
        expect(assets[1]).toMatchSnapshot();
      });

      test('True (No Manifest)', async () => {
        const config = {
          loader: {
            test: /\.babylon$/,
            options: {
              emitFile: true,
            },
          },
        };

        const stats = await webpack('noManifest/fixture.js', config);
        const { assets } = stats.toJson().modules[1];

        expect(assets[0]).toMatchSnapshot();
      });

      test('False', async () => {
        const config = {
          loader: {
            test: /\.babylon$/,
            options: {
              emitFile: false,
            },
          },
        };

        const stats = await webpack('fixture.js', config);
        const { assets } = stats.toJson().modules[1];

        expect(assets[0]).toBe(undefined);
        expect(assets[1]).toBe(undefined);
      });
    });
  });
});
