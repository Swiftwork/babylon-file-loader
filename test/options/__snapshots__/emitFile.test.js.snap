// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options emitFile {Boolean} True (Default) 1`] = `"77f94cfeb835c099e9eeab739989952d.babylon"`;
exports[`Options emitFile {Boolean} True (Default) 2`] = `"77f94cfeb835c099e9eeab739989952d.babylon.manifest"`;

exports[`Options emitFile {Boolean} True (No Manifest) 1`] = `"95d20d8b215d33fb77af8cf2120e3208.babylon"`;
